src/amo/components/SearchContextCard/index.js

```diff
@@ -25,10 +31,37 @@ export class SearchContextCardBase extends React.Component {
   render() {
     const { count, filters, i18n, loading } = this.props;
     const { query } = filters;
+    const { addonType } = filters;
 
     let searchText;
 
-    if (!loading && query) {
+    if (!loading && query && addonType) {
+      if (addonType === ADDON_TYPE_EXTENSION) {
+        searchText = i18n.sprintf(i18n.ngettext(
+          '%(count)s extension found for "%(query)s"',
+          '%(count)s extensions found for "%(query)s"',
+          count), { count: i18n.formatNumber(count), query }
+        );
+      } else if (addonType === ADDON_TYPE_DICT) {
+        searchText = i18n.sprintf(i18n.ngettext(
+          '%(count)s dictionary found for "%(query)s"',
+          '%(count)s dictionaries found for "%(query)s"',
+          count), { count: i18n.formatNumber(count), query }
+        );
+      } else if (addonType === ADDON_TYPE_LANG) {
+        searchText = i18n.sprintf(i18n.ngettext(
+          '%(count)s language pack found for "%(query)s"',
+          '%(count)s language packs found for "%(query)s"',
+          count), { count: i18n.formatNumber(count), query }
+        );
+      } else if (addonType === ADDON_TYPE_THEME) {
+        searchText = i18n.sprintf(i18n.ngettext(
+          '%(count)s theme found for "%(query)s"',
+          '%(count)s themes found for "%(query)s"',
+          count), { count: i18n.formatNumber(count), query }
+        );
+      }
```

Comment:
This could use an `else` block that shows some kind of generic message like '12 results found for...'